python/context.cpp

```diff
@@ -153,8 +153,8 @@ void register_contexts(pybind11::module& m) {
 #else
         .def(pybind11::init(
             [](int threads, pybind11::object gpu){
-                auto gpu_id = py2optional<int>(gpu,
-                        "gpu id must be None, or a non-negative integer.", is_nonneg_int);
+                const char* gpu_err_str = "gpu_id must be None, or a non-negative integer";
+                auto gpu_id = py2optional<int>(gpu, gpu_err_str, is_nonneg_int);
```

Comment:
having the error string inline here doesn't make the code harder to read.

Reply:
Changed back.